### ISSUE TITLE
Help Wanted

### ISSUE THREAD

**Issue by bmulholland** [nuxt-community/auth-module#1393]
As most have noticed, and [previously flagged](https://github.com/nuxt-community/auth-module/issues/893), this project needs more attention. I have just put in a few days over Christmas to upgrade all the packages and ensure the demo app is working, but still improvements are needed. This is all table stakes for a maintained app, and yet I have limited time to do it myself. I am asking for help with any of these tasks, which would be the most immediately helpful in moving the project toward a more stable and bug-free (hah!) state.

If you make any improvement on any of these, even partially, please create a PR and assign me for review.

Happy to answer questions, or even make time for a short call to discuss anything.

Tasks:

- [x] Get https://github.com/nuxt-community/auth-module/pull/1306 green
- [ ] In dev-mode demo app (`yarn dev`), fix Login with Google
- [ ] Write instructions for setting up the dev-mode demo app (`yarn dev`) to test Facebook
- [ ] In dev-mode demo app (`yarn dev`), fix Login with Github
- [ ] Write instructions for setting up the dev-mode demo app (`yarn dev`) to test Laravel (Sanctum, JWT, Passport, and Password Grant -- all at once, or one at a time)
- [ ] Improve test coverage: even just one test at a time would be appreciated
- [ ] Make it simpler to add tests: write documentation on how they work, refactor so there's less boilerplate
- [ ] Test, report, and fix issues with nuxt bridge and nuxt 3: https://github.com/nuxt-community/auth-module/issues/1323
- [ ] Review TODOs/FIXMEs in the app and either: create a PR with a fix or removing it (with justification), or create a Github Issue with next steps

**Comment by Intevel** [comment]
Github Auth works for me on the demo page. @bmulholland

**Comment by Intevel** [comment]
Login with Google also works for me on the demo page.

**Comment by bmulholland** [comment]
@Intevel Thanks for the help! What about when run locally with `yarn dev`?

**Comment by Intevel** [comment]
> @Intevel Thanks for the help! What about when run locally with `yarn dev`?

Oh no, I have now tested only Github but it does not work or I get a 404 error on Github? Do you mean that? @bmulholland

**Comment by bmulholland** [comment]
Yep, that's what I mean. We need to be able to test changes locally for new development/changes/refactors.

**Comment by Intevel** [comment]
Yes that would be good, I just wonder why we get a 404 error from Github.

**Comment by Intevel** [comment]
I got it, it doesn't include the client ID in the params.

**Comment by Intevel** [comment]
Okay, it does not add the `client_id` to the paramenters. That's why we get a 404 error

**Comment by Intevel** [comment]
But Google works in the Development Mode of the demo. @bmulholland

**Comment by Intevel** [comment]
I reloaded the project and wrote my oAuth2 data directly into the config without `.env`, Github works too.

**Comment by Intevel** [comment]
Sorry for commenting so much right now, I'm trying to understand. So I don't get **404** error anymore, as long as I write the data directly to `nuxt.config.ts` everything works, as soon as I use `.env` the callback doesn't work anymore and it loads all the time n

**Comment by Intevel** [comment]
Now I don't understand anything anymore, Google doesn't work now either. 😨

**Comment by Intevel** [comment]
okay, that's stupid. Github now it works again after I restarted the demo.

**Comment by bmulholland** [comment]
Could you please create a specific ticket to track this work and document the results of your experiments into the README as a PR? I can help clean up if needed. That will help keep this specific conversation separate from the broader Help Wanted topic.

**Comment by Intevel** [comment]
What do you mean by "ticket"?

**Comment by andrewdleach** [comment]
@bmulholland, does this work help us get `auth` working with Nuxt3? (i'm historically BE oriented, but am more than happy to help out if more hands are needed)

**Comment by bmulholland** [comment]
No, the nuxt 3 version is a complete rewrite

**Comment by andrewdleach** [comment]
Do you happen to know where that work is being tracked @bmulholland? I'm new-er to Nuxt development and would love any pointers to docs that you find helpful

**Comment by bmulholland** [comment]
It's being worked on privately first

**Comment by th1nkgr33n** [comment]
Hey @bmulholland believing rumors the nuxt3 migration happens in a private beta :D ?
My Team and I need to make some decisions and maybe you are willing to give us some hints.
At the point we have an nuxt2 project (using this auth module) as well as a new nuxt3 project starting now.

The nuxt3 project needs oauth as well and we have looked forward that this module would be part of the stable release.
Things often don't turn out the way you planned, but can you tell me when this module would be migrated?
Otherwise we need to choose another one (maybe just as interim solution) and come back later to this, but I only want to do this, if you are telling me for example that your hitting the finish linie not earlier than in a couple of weeks/months.

On the other hand we have a team internal discussion around to support the migration process. Is this possible?

Thanks for response and have a nice day!

**Comment by bmulholland** [comment]
I'm not involved in the rewrite and, as far as I am aware, there's no clear timeline. My company is also -- in part -- blocked from upgrading to Nuxt 3 because of this module. Our plans to upgrade include migrating off of nuxt/auth.

**Comment by th1nkgr33n** [comment]
@bmulholland thanks for response! Good luck for migration!
